modules/channel-twilio/src/backend/client.ts

```diff
@@ -0,0 +1,265 @@
+import * as sdk from 'botpress/sdk'
+import LRUCache from 'lru-cache'
+import { Twilio, validateRequest } from 'twilio'
+
+import { Config } from '../config'
+
+import { Clients, MessageOption, TwilioRequestBody } from './typings'
+
+const MIDDLEWARE_NAME = 'twilio.sendMessage'
+
+export class TwilioClient {
+  private logger: sdk.Logger
+  private twilio: Twilio
+  private webhookUrl: string
+  private kvs: sdk.KvsService
+
+  constructor(
+    private bp: typeof sdk,
+    private botId: string,
+    private config: Config,
+    private router: sdk.http.RouterExtension,
+    private route: string
+  ) {
+    this.logger = bp.logger.forBot(botId)
+  }
+
+  async initialize() {
+    if (!this.config.accountSID || !this.config.authToken) {
+      return this.logger.error(`[${this.botId}] The accountSID and authToken must be configured to use this channel.`)
+    }
+
+    const url = (await this.router.getPublicPath()) + this.route
+    this.webhookUrl = url.replace('BOT_ID', this.botId)
+
+    this.twilio = new Twilio(this.config.accountSID, this.config.authToken)
+    this.kvs = this.bp.kvs.forBot(this.botId)
+
+    this.logger.info(`Twilio webhook listening at ${this.webhookUrl}`)
+  }
+
+  auth(req): boolean {
+    const signature = req.headers['x-twilio-signature']
+    return validateRequest(this.config.authToken, signature, this.webhookUrl, req.body)
+  }
+
+  async handleWebhookRequest(body: TwilioRequestBody) {
+    const to = body.To
+    const from = body.From
+    const text = body.Body
+
+    const index = Number(text)
+    if (index && (await this.handleIndexReponse(index - 1, from, to))) {
+      return
+    }
```

Comment:
What do you think of changing the payload instead of having different handlers ? It would remove the duplicate sendEvent methods,. Ex:

```js
let payload = { type: 'text', text }
if(index){
  const numberResponse = await this.handleResponse(index, target, threadId)
  payload = numberResponse ?? payload
}
```